base_multi_image/hooks.py

```diff
@@ -75,11 +75,73 @@ def uninstall_hook_for_submodules(cr, registry, model):
     :param str model:
         Model technical name, like "res.partner". All multi-images for that
         model will be deleted
+    
+    :param str field:
+        Binary field that had the images in that :param:`model`, like
+        ``image``.
+    
+    :param str field_medium:
+        Binary field that had the medium-sized images in that :param:`model`, like
+        ``image_medium``.
+    
+    :param str field_small:
+        Binary field that had the small-sized images in that :param:`model`, like
+        ``image_small``.
     """
     env = api.Environment(cr, SUPERUSER_ID, {})
     with cr.savepoint():
         Image = env["base_multi_image.image"]
-        images = Image.search([("owner_model", "=", model)])
+        images = Image.search([("owner_model", "=", model)], order="sequence, id")
+        if images and (field or field_medium or field_small):
+            main_images = {}
+            for image in images:
+                if image.owner_id not in main_images:
+                    main_images[image.owner_id] = image
+            main_images = main_images.values()
+            Model = env[model]
+            Field = field and Model._fields[field]
+            FieldMedium = field_medium and Model._fields[field_medium]
+            FieldSmall = field_small and Model._fields[field_small]
```

Comment:
one of these 3 can fail with `KeyError` easily

Reply:
Hi @Yajo , if a KeyError is raised it means that the uninstall hook hasn't been used correctly by the module that depends on it. It would be a programming error.

Reply:
In such case, the arguments shouldn't be optional

Reply:
If the new parameters would be required then it will break existing modules using the hook.

They should be optional because the introduced behavior isn't the desired in some cases as product variants in OCA/product-attribute#359